File: catkin_ws/src/my_robot_tutorials/scripts/number_counter.py
#!/usr/bin/env python3

import rospy 
from std_msgs.msg import Int64
from std_srvs.srv import SetBool

counter = 0
pub = None

def callback_add_ints(msg):
    global counter
    counter += msg.data
    rospy.loginfo("The answer of addtion : ")
    rospy.loginfo(counter)
    answer = Int64()
    answer.data = counter
    pub.publish(answer)

#practice about service
def callback_reset_counter(req):
    if req.data == 1:
        global counter
        counter = 0
        rospy.loginfo("Counter has benn reset!!")
        return True, "counter has been successfully reset"
    else:
        return False, "counter has not been reset"


if __name__ == '__main__':
    rospy.init_node("number_count")
    sub = rospy.Subscriber('/number', Int64, callback_add_ints)
    pub = rospy.Publisher('/number_count', Int64, queue_size= 10)
    #rospy.spin()

    #practice about service
    reset_service = rospy.Service('/reset_counter',SetBool, callback_reset_counter)

    rospy.spin()

File: catkin_ws/src/my_robot_tutorials/src/number_counter.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <std_srvs/SetBool.h>

int counter = 0;
ros::Publisher pub;

void callback_add_Ints(const std_msgs::Int64& msg){
    counter += msg.data;
    ROS_INFO("The answer of addtion : %i", counter);
    std_msgs::Int64 answer;
    answer.data = counter;
    pub.publish(answer);
}

//practice for ros service
bool callback_reset_counter(std_srvs::SetBool::Request &req,
                            std_srvs::SetBool::Response &res)
{
    if (req.data == 1){
        counter = 0;
        ROS_INFO("Counter has been reset!!!!");
        res.success = true;
        res.message = "Counter has been successfully reset";
    }
    else {
        res.success = true;
        res.message = "Counter has not been successfully reset";
    }
    return true;
}


int main(int argc, char **argv){
    ros::init(argc, argv, "number_count");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/number",1000, callback_add_Ints);
    pub = nh.advertise<std_msgs::Int64>("/number_count",10);

    //practice for ros Service

    ros::ServiceServer server = nh.advertiseService("/reset_counter", callback_reset_counter);

    ros::spin();
}